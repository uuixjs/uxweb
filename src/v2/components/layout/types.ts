// from inject-layout
import { BorderRadius, BorderRadiusValues } from "lib/ui-utils";

/**
 * Many string values in this file must match the CSS values
 * for the corresponding property because they get used directly!
 */

export enum AlignContent {
  Start = "flex-start",
  End = "flex-end",
  Center = "center",
  Between = "space-between",
  Around = "space-around",
  Stretch = "stretch",
}

export type AlignContentValue =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "stretch"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum AlignItems {
  Start = "flex-start",
  End = "flex-end",
  Center = "center",
  Baseline = "baseline",
  Stretch = "stretch",
}

export type AlignItemsValue =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum AlignSelf {
  Auto = "auto",
  Start = "flex-start",
  End = "flex-end",
  Center = "center",
  Baseline = "baseline",
  Stretch = "stretch",
}

export type AlignSelfValue =
  | "auto"
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum Cursor {
  Default = "default",
  Auto = "auto",
  NotAllowed = "not-allowed",
  Pointer = "pointer",
}

export type CursorValue = "default" | "auto" | "not-allowed" | "pointer"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum Display {
  Block = "block",
  Flex = "flex",
  Initial = "initial",
  Inline = "inline",
  InlineBlock = "inline-block",
  InlineFlex = "inline-flex",
  Hide = "none",
  HideAccessible = "hide-accessible",
}

export type DisplayValue =
  | "block"
  | "flex"
  | "initial"
  | "inline"
  | "inline-block"
  | "inline-flex"
  | "none"
  | "hide-accessible"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum FlexDirection {
  Column = "column",
  ColumnReverse = "column-reverse",
  Row = "row",
  RowReverse = "row-reverse",
}

export type FlexDirectionValue =
  | "column"
  | "column-reverse"
  | "row"
  | "row-reverse"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum FlexWrap {
  Wrap = "wrap",
  NoWrap = "nowrap",
  WrapReverse = "wrap-reverse",
}

export type FlexWrapValue = "wrap" | "nowrap" | "wrap-reverse"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum JustifyContent {
  Start = "flex-start",
  End = "flex-end",
  Center = "center",
  Between = "space-between",
  Around = "space-around",
}

export type JustifyContentValue =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum Overflow {
  Auto = "auto",
  Scroll = "scroll",
  Visible = "visible",
  Hidden = "hidden",
}

export type OverflowValue = "auto" | "scroll" | "visible" | "hidden"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum Position {
  Relative = "relative",
  Absolute = "absolute",
  Fixed = "fixed",
}

export type PositionValue = "relative" | "absolute" | "fixed"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum Resize {
  None = "none",
  X = "horizontal",
  Y = "vertical",
}

export type ResizeValue = "none" | "horizontal" | "vertical"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum TextAlign {
  Left = "left",
  Center = "center",
  Right = "right",
  Justify = "justify",
}

export type TextAlignValue = "left" | "center" | "right" | "justify"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum VerticalAlign {
  Top = "top",
  Middle = "middle",
  Baseline = "baseline",
  Bottom = "bottom",
  TextTop = "text-top",
  TextBottom = "text-bottom",
}

export type VerticalAlignValue =
  | "top"
  | "middle"
  | "baseline"
  | "bottom"
  | "text-top"
  | "text-bottom"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum Visibility {
  Visible = "visible",
  Hidden = "hidden",
}

export type VisibilityValue = "visible" | "hidden"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum ZIndex {
  Default = "z-index-default",
  Above = "z-index-above",
  Below = "z-index-below",
}

// `ZIndexValue` is not defined to avoid possible confusion; only the Enum is accepted for now.

export type FlexValue = 0 | 1 | 2 | 3 | 4;

// from inject-styled-layout

export enum Background {
  Base = "base",
  Alt = "alt",
  Alt2 = "alt-2",
  Overlay = "overlay",
  Accent = "accent",
  AccentAlt = "accent-alt",
  AccentAlt2 = "accent-alt-2",
  Inherit = "inherit",
}

export type BackgroundValue =
  | "base"
  | "alt"
  | "alt-2"
  | "overlay"
  | "accent"
  | "accent-alt"
  | "accent-alt-2"
  | "inherit"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export enum Color {
  Base = "base",
  Alt = "alt",
  Alt2 = "alt-2",
  Link = "link",
  Live = "live",
  Error = "error",
  Overlay = "overlay",
  OverlayAlt = "overlay-alt",
  Inherit = "inherit",
}

export type ColorValue =
  | "base"
  | "alt"
  | "alt-2"
  | "link"
  | "live"
  | "error"
  | "overlay"
  | "overlay-alt"
  | "inherit"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export type Elevation = 0 | 1 | 2 | 3 | 4 | 5;

export enum FontSize {
  Size1 = "font-size-1",
  Size2 = "font-size-2",
  Size3 = "font-size-3",
  Size4 = "font-size-4",
  Size5 = "font-size-5",
  Size6 = "font-size-6",
  Size7 = "font-size-7",
  Size8 = "font-size-8",
}

// `FontSizeValue` is not defined to avoid possible confusion; only the Enum is accepted for now.

export enum FontWeight {
  Regular = "regular",
  SemiBold = "semibold",
  Bold = "bold",
}

export type FontWeightValue = "regular" | "semibold" | "bold"; // TODO: https://jira.xarth.tv/browse/COREUI-3554

export type BorderRadiusProps = BorderRadius | BorderRadiusValues;
