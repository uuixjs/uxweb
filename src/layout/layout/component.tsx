import { CommonLayoutProps, propMap } from "../utils/layout-props";
import { DetailedHTMLProps, HTMLAttributes } from "react";
import { StyledComponent, styled } from "lib";

import { getLayoutStyles } from "../utils/common";

type DivProps = DetailedHTMLProps<
  HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

export interface LayoutProps
  extends CommonLayoutProps,
    // Omit DivProps props which are already defined as part of CommonLayoutProps
    Omit<DivProps, "color" | "role" | "ref"> {}

export const Layout = styled.div.withConfig({
  shouldForwardProp: (propName, defaultValidatorFn) => {
    if (propName in propMap) {
      return false;
    }

    return defaultValidatorFn(propName);
  },
})<LayoutProps>((props) => getLayoutStyles(props, propMap)) as StyledComponent<
  "div",
  never, // Prevent the "theme" prop from accepting any value
  LayoutProps,
  never
>;

Layout.displayName = "Layout";
