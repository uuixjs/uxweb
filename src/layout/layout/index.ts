export { Layout, LayoutProps } from "./component";
