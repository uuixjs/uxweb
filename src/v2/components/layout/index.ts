export * from "./inject-layout";
export * from "./inject-styled-layout";
export * from "./layout";
export * from "./styled-layout";
export * from "./types";
