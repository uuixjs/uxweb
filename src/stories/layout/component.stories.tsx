import * as React from "react";

import {
  AlignItems,
  Background,
  Color,
  CoreText,
  Cursor,
  JustifyContent,
  Layout,
  LayoutProps,
  TextType,
} from "v2";
import {
  CombinationGenerator,
  CombinationMode,
} from "../components/combination-generator";

import { OverlayPreview } from "../components/overlay-preview";

export default { title: "Layout" };

export const cursor = () => (
  <>
    <Layout cursor={Cursor.Default} padding={1}>
      Hovering over this element should display an arrow cursor.
    </Layout>
    <Layout cursor={Cursor.Auto} padding={1}>
      Hovering over this element should display a text cursor.
    </Layout>
    <Layout cursor={Cursor.NotAllowed} padding={1}>
      Hovering over this element should display the "not-allowed" cursor.
    </Layout>
    <Layout cursor={Cursor.Pointer} padding={1}>
      Hovering over this element should display the pointer cursor.
    </Layout>
  </>
);

const commonProps: LayoutProps = {
  padding: 5,
  justifyContent: JustifyContent.Center,
  alignItems: AlignItems.Center,
};

export const examples = () => {
  return (
    <Layout margin={{ bottom: 2 }}>
      <CoreText type={TextType.H2}>Layout Examples</CoreText>

      <CombinationGenerator
        mode={CombinationMode.Simple}
        fields={[
          {
            propKey: "color",
            propValues: [
              Color.Base,
              Color.Alt,
              Color.Alt2,
              Color.Error,
              Color.Inherit,
              Color.Link,
              Color.Live,
              Color.Overlay,
              Color.OverlayAlt,
            ],
          },
          {
            propKey: "background",
            propValues: [
              Background.Accent,
              Background.AccentAlt,
              Background.AccentAlt2,
              Background.Alt,
              Background.Alt2,
              Background.Base,
              Background.Overlay,
            ],
          },
        ]}
      >
        <Layout {...commonProps}>Text</Layout>
      </CombinationGenerator>
    </Layout>
  );
};

export const withOverlay = () => {
  return (
    <Layout margin={{ bottom: 2 }}>
      <CoreText type={TextType.H3}>Overlay Mode:</CoreText>
      <OverlayPreview>
        <CombinationGenerator
          mode={CombinationMode.Simple}
          fields={[
            {
              propKey: "color",
              propValues: [Color.Overlay, Color.OverlayAlt],
            },
            {
              propKey: "background",
              propValues: [
                Background.Accent,
                Background.AccentAlt,
                Background.AccentAlt2,
                Background.Overlay,
              ],
            },
          ]}
        >
          <Layout {...commonProps}>Text</Layout>
        </CombinationGenerator>
      </OverlayPreview>
    </Layout>
  );
};

export const withElevation = () => {
  return (
    <Layout margin={{ bottom: 2 }}>
      <CoreText type={TextType.H2}>Layout Examples</CoreText>

      <CombinationGenerator
        mode={CombinationMode.Simple}
        fields={[
          {
            propKey: "color",
            propValues: [Color.Base],
          },
          {
            propKey: "background",
            propValues: [Background.Alt, Background.Alt2, Background.Base],
          },
        ]}
      >
        <Layout {...commonProps} elevation={0}>
          Text
        </Layout>
        <Layout {...commonProps} elevation={1}>
          Text
        </Layout>
        <Layout {...commonProps} elevation={2}>
          Text
        </Layout>
        <Layout {...commonProps} elevation={3}>
          Text
        </Layout>
        <Layout {...commonProps} elevation={4}>
          Text
        </Layout>
        <Layout {...commonProps} elevation={5}>
          Text
        </Layout>
      </CombinationGenerator>
    </Layout>
  );
};
