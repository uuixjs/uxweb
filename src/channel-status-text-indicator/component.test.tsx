import { setupShallowTest } from "../../tests/helpers";
import { Background } from "../layout";
import {
  ChannelStatusTextIndicatorComponent,
  ChannelStatusTextIndicatorProps,
  ChannelStatusTextIndicatorSize,
  ChannelStatusTextIndicatorType,
} from "./component";

function getRequiredProps(): ChannelStatusTextIndicatorProps {
  return {
    label: "Test label",
  };
}

function getOptionalProps(): ChannelStatusTextIndicatorProps {
  return {
    label: "Test label",
    size: ChannelStatusTextIndicatorSize.Large,
    type: ChannelStatusTextIndicatorType.Hosting,
    overlay: true,
    mask: Background.AccentAlt2,
  };
}

const setupShallow = setupShallowTest(ChannelStatusTextIndicatorComponent);

describe("ChannelStatusTextIndicator", () => {
  it("renders channel status text indicator component with required props", () => {
    const { wrapper } = setupShallow(getRequiredProps());
    expect(wrapper).toMatchSnapshot();
  });

  it("renders channel status text indicator component with all props", () => {
    const { wrapper } = setupShallow(getOptionalProps());
    expect(wrapper).toMatchSnapshot();
  });
});
