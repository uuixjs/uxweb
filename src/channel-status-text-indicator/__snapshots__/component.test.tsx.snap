// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelStatusTextIndicator renders channel status text indicator component with all props 1`] = `
<Styled(Layout)
  background="accent-alt-2"
  borderRadius="border-radius-medium"
  display="inline-block"
>
  <styled.div
    className="tw-channel-status-text-indicator"
    colorMap={
      Object {
        "fill": "color-fill-channel-status-text-indicator-hosting",
        "overlay": Object {
          "fill": "color-fill-channel-status-text-indicator-hosting-overlay",
          "text": "color-text-channel-status-text-indicator-hosting-overlay",
        },
        "text": "color-text-channel-status-text-indicator-hosting",
      }
    }
    fontSize="font-size-5"
    overlay={true}
    padding="0.3rem 1rem"
  >
    <CoreText
      bold={true}
      transform="uppercase"
      whiteSpace="nowrap"
    >
      Test label
    </CoreText>
  </styled.div>
</Styled(Layout)>
`;

exports[`ChannelStatusTextIndicator renders channel status text indicator component with required props 1`] = `
<styled.div
  className="tw-channel-status-text-indicator"
  colorMap={
    Object {
      "fill": "color-fill-live",
      "text": "color-text-overlay",
    }
  }
  fontSize="font-size-6"
  padding="0 0.5rem"
>
  <CoreText
    bold={true}
    transform="uppercase"
    whiteSpace="nowrap"
  >
    Test label
  </CoreText>
</styled.div>
`;
